extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientBuildConfig.java

```diff
@@ -0,0 +1,148 @@
+package io.quarkus.kubernetes.client.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "kubernetes-client", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class KubernetesClientBuildConfig {
+
+    /**
+     * Whether or not the client should trust a self signed certificate if so presented by the API server
+     */
+    @ConfigItem(defaultValue = "false")
+    boolean trustCerts;
+
+    /**
+     * URL of the Kubernetes API server
+     */
+    @ConfigItem
+    String masterUrl;
+
+    /**
+     * Default namespace to use
+     */
+    @ConfigItem
+    String namespace;
+
+    /**
+     * CA Certificate file
+     */
+    @ConfigItem
+    String caCertFile;
+
+    /**
+     * CA Certificate data
+     */
+    @ConfigItem
+    String caCertData;
+
+    /**
+     * Client certificate file
+     */
+    @ConfigItem
+    String clientCertFile;
+
+    /**
+     * Client certificate data
+     */
+    @ConfigItem
+    String clientCertData;
+
+    /**
+     * Client key file
+     */
+    @ConfigItem
+    String clientKeyFile;
+
+    /**
+     * Client key data
+     */
+    @ConfigItem
+    String clientKeyData;
+
+    /**
+     * Client key algorithm
+     */
+    @ConfigItem
+    String clientKeyAlgo;
+
+    /**
+     * Client Key passphrase
+     */
+    @ConfigItem
+    String clientKeyPassphrase;
+
+    /**
+     * Kubernetes auth username
+     */
+    @ConfigItem
+    String username;
+
+    /**
+     * Kubernetes auth password
+     */
+    @ConfigItem
+    String password;
+
+    /**
+     * Watch reconnect interval in milliseconds
```

Comment:
Shouldn't this be a Duration instead?

Same for all the intervals/timeouts.

Reply:
That is a good idea. I tried that initially and failed because of empty default values, but I should definitely reexamine now that I have added default values

Reply:
Implemented, thanks for pointing it out!

Reply:
If a default value is empty, it should be an `Optional<Duration>`.

Reply:
I'll keep that in mind. For this PR I just used the proper default values